utils/src/server/rpc_module.rs

```diff
@@ -343,19 +348,85 @@ impl SubscriptionSink {
 		self.inner_send(msg).map_err(Into::into)
 	}
 
-	fn inner_send(&self, msg: String) -> Result<(), Error> {
-		if self.is_online() {
-			self.inner.unbounded_send(msg).map_err(|e| Error::Internal(e.into_send_error()))
+	fn inner_send(&mut self, msg: String) -> Result<(), Error> {
+		let res = if let Some(online) = self.keep_alive.as_ref() {
+			if online.is_canceled() {
+				return Err(subscription_closed_by_client());
```

Comment:
This early return here, it means we don't run lines 365-367 to close the `KeepAlive`, why is that?